internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -74,7 +80,26 @@ func (r *NumaflowControllerRolloutReconciler) Reconcile(ctx context.Context, req
 		}
 	}
 
-	numaLogger.Info(string(numaflowControllerRollout.Spec.Controller.Version))
+	// TODO: need to generate the correct spec and setup the appropriate meta fields
+	// based on the NumaflowControllerRollout spec.controller.version
+	obj := kubernetes.GenericObject{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "",
+			APIVersion: "",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      numaflowControllerRollout.Name,
+			Namespace: "numaflow-system",
+		},
+		Spec: runtime.RawExtension{},
+	}
+
+	// TODO: what is the plural of this kind?
+	err := kubernetes.UpdateCRSpec(ctx, r.restConfig, &obj, "<plural>")
```

Comment:
it's "interstepbufferservices"

You can see it if you do `kubectl get crd interstepbufferservices.numaflow.numaproj.io -o yaml`